benchmarks/rays/rays.cpp

```diff
@@ -0,0 +1,187 @@
+/****************************************************************************
+ * Copyright (c) 2017-2022 by the ArborX authors                            *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "ArborX_Predicates.hpp"
+#include <ArborX_DistributedTree.hpp>
+#include <ArborX_Version.hpp>
+
+#include <Kokkos_Core.hpp>
+
+#include <fstream>
+
+#include <mpi.h>
+
+using ExecutionSpace = Kokkos::DefaultExecutionSpace;
+using MemorySpace = ExecutionSpace::memory_space;
+
+constexpr int n_rays = 307200;
+
+template <typename MemorySpace>
+std::pair<Kokkos::View<ArborX::Experimental::Ray[n_rays], MemorySpace>,
+          Kokkos::View<double[n_rays], MemorySpace>>
+read_data()
+{
+  std::string filename = "rays_cam-0-0_test_full.csv";
+  Kokkos::View<ArborX::Experimental::Ray[n_rays], MemorySpace> rays("rays");
+  auto rays_host = Kokkos::create_mirror_view(rays);
+  Kokkos::View<double[n_rays], MemorySpace> temperatures("temperatures");
+  auto temperatures_host = Kokkos::create_mirror_view(temperatures);
+
+  std::ifstream file;
+  file.open(filename);
```

Comment:
```suggestion
  std::ifstream file(filename);
```
You would want to have a check for failure.